GoMeet/src/main/java/com/google/sps/servlets/PopularLocationServlet.java

```diff
@@ -0,0 +1,60 @@
+package main.java.com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.EntityNotFoundException;
+import com.google.gson.Gson;
+import com.google.sps.data.ServletUtil;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Comparator;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import main.java.com.google.sps.dao.Dao;
+import main.java.com.google.sps.data.Location;
+import main.java.com.google.sps.dao.LocationDao;
+import org.jsoup.Jsoup;
+import org.jsoup.safety.Whitelist;
+
+/** Handles fetching and saving location data. */
+@WebServlet("/popular-location-data")
+public class PopularLocationServlet extends HttpServlet {
+  private Dao<Location> locationDao = new LocationDao();
+
+  /** Responds with a JSON array containing the most popular location data. */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("application/json");
+
+    List<Location> popularLocations = getPopularLocation();
+    String json = ServletUtil.convertToJson(popularLocations);
+    response.getWriter().println(json);
+  }
+
+  /** Returns a list of the locations with the highest vote count. */
+  private List<Location> getPopularLocation() {
+    List<Location> popularLocations = new ArrayList<>();
+    List<Location> allLocations = locationDao.getAll();
+
+    int maxVoteCount = 0;
+    for (Location location : allLocations) {
+      int currVoteCount = location.getVoteCount();
+      if (currVoteCount < maxVoteCount) {
+        continue;
+      }
+      if (currVoteCount > maxVoteCount) {
+        popularLocations.clear();
+        maxVoteCount = currVoteCount;
+      }
+      popularLocations.add(location);
+    }
+    return popularLocations;
+  }
```

Comment:
Kind of confused by how this function works. Why do you clear the whole popularLocations arraylist each time the current iteration has a higher vote count? How many locations do you expect to be included in the final result of popularLocations (e.g. top 3 most popular locations)? Might be good to explain this in a comment above the function.

Reply:
So for this function, if the locations stored are: 
- Location A, VoteCount: 5
- Location B, VoteCount: 5
- Location C, VoteCount: 1
Then Location A and Location B are returned in the list. 

There is not expected number of popular locations. It is is the number of locations in the database with the highest vote count. This can be multiple locations. 

I'll update the Javadoc to make it clearer.

Reply:
Ok that makes sense